### DISCUSSION TITLE
How to get old dynamic input value and input number count

### DISCUSSION THREAD

**Discussion by AsemAbbas** [livewire/livewire#4208] [1. Help]
### **This is my create and edit functions after create  and when hit the edit button I want the value and return input number count**

![image](https://user-images.githubusercontent.com/80175112/141614276-e7a257a1-1bae-485a-a7b4-ed5610888be0.png)

### **The edit functions return just first value not all value**

![image](https://user-images.githubusercontent.com/80175112/141614310-71f9bbf3-bab1-41da-9fb1-2cd5e083aff9.png)

### **But in database  have 3**

![image](https://user-images.githubusercontent.com/80175112/141614470-7f7e33ca-2f1f-4cfb-b697-b7ec99d867fa.png)

**Comment by ralphjsmit** [comment]
Hey Asem, I see that you access the `terms` as an array, so like this: `$arrested['terms']`. You have added the Term model as an Eloquent relationship (you are using `terms()->create()`), you should access it as `$arrested->terms` or `$arrested->terms-()>get()`.

Could you try that and see if it works?

**Reply by AsemAbbas** [reply]
no its not work :(

**Comment by zepfietje** [comment]
@AsemAbbas could you elaborate on the exact problem you're experiencing?

> The edit functions return just first value not all value

You call `dd()` which dumps the value and stops execution.
That's why you see just a single (the first) value printed instead of all values you loop through.

**Reply by AsemAbbas** [reply]
but I have three in my database not just one

**Reply by ralphjsmit** [reply]
What Zep means is that `dd()` stops execution. That means that it stops after it has printed the first value. So if there are three total values, and it stops after printing the first value, you'll only see the first value.

Try using `dump()` instead of `dd()`.

**Reply by AsemAbbas** [reply]
Yes, it returned three values now
so can I use count($term) to return it for $steps and it will show 3 input??

### **I try it but its not work**

![image](https://user-images.githubusercontent.com/80175112/141650096-2c24c7d1-a978-4ae6-8c9f-310d999116f4.png)

**Reply by AsemAbbas** [reply]
I want return how many value I have to `$this->steps` but when I edit

![image](https://user-images.githubusercontent.com/80175112/141650227-6e46b23b-f342-4778-8b83-d7e391c7b9cd.png)

![image](https://user-images.githubusercontent.com/80175112/141650267-fdb9becc-e6d2-4b09-a4b2-d8e74e23b116.png)

**Reply by zepfietje** [reply]
> Yes, it returned three values now so can I use count($term) to return it for $steps and it will show 3 input??
> 
> ### **I try it but its not work**
> ![image](https://user-images.githubusercontent.com/80175112/141650096-2c24c7d1-a978-4ae6-8c9f-310d999116f4.png)

If I understand correctly what you want to achieve, that should be:
```php
$this->steps = count($arrested['terms']);
```

**Reply by AsemAbbas** [reply]
![image](https://user-images.githubusercontent.com/80175112/141652251-734b8dc4-18c9-4813-a4ed-316374a82e74.png)


### **its return just 1 input**

![image](https://user-images.githubusercontent.com/80175112/141652250-589824c4-ee66-4797-bde8-138c8bcf2dac.png)

**Reply by zepfietje** [reply]
> If I understand correctly what you want to achieve, that should be:
> 
> ```
> $this->steps = count($arrested['terms']);
> ```

Have you tried my suggestion?

The way you're currently assigning to `$this->steps` would lead it to contain a value as follows (when you have 3 items like you said):
```php
[
    3,
]
```

**Reply by AsemAbbas** [reply]
### **when I am use this** 
` $this->steps = count($arrested['terms']); `

### **it return this error**

![image](https://user-images.githubusercontent.com/80175112/141675577-8ac8b490-ece8-4a7d-bcee-544312a9e08f.png)

### **but I solve it like this**

![image](https://user-images.githubusercontent.com/80175112/141675590-bbc38416-9757-4fd2-bd4a-2d7fceba4af6.png)

### **after this its return nothing**

![image](https://user-images.githubusercontent.com/80175112/141675677-e62a75c0-b1e6-4757-a811-7403f36153fb.png)

### **and when click to add input its return this error**

![image](https://user-images.githubusercontent.com/80175112/141675700-439e4d98-fae9-4fcd-b6bf-947c95b2d975.png)

**Reply by zepfietje** [reply]
> ### **when I am use this**
> `$this->steps = count($arrested['terms']);`
> 
> ### **it return this error**
> ![image](https://user-images.githubusercontent.com/80175112/141675577-8ac8b490-ece8-4a7d-bcee-544312a9e08f.png)

That's because you try to loop over `$steps` with `@foreach`, while you should be using a `@for` loop if you want `$steps` to contain an integer number of steps.

> ### **but I solve it like this**
> ![image](https://user-images.githubusercontent.com/80175112/141675590-bbc38416-9757-4fd2-bd4a-2d7fceba4af6.png)
> 
> ### **after this its return nothing**
> ![image](https://user-images.githubusercontent.com/80175112/141675677-e62a75c0-b1e6-4757-a811-7403f36153fb.png)

That's obvious, since `$steps` contains an integer like I said above.

I think it would be good if you explained the exact functionality you want to achieve. Currently it's really hard for me to help if it's all this unclear. If you work it out really clearly and detailed, others will be able to help you much more easily.

**Comment by AsemAbbas** [comment]
### **The create function is ok but I want when I am click on edit button return how many input  I should have and  the old value in this inputs**

**View**
```
<div class="form-group col-md-6">
    <label>البنود الفرعية</label>
    <i class="fa fa-plus-circle text-primary" wire:click="increment"></i>
    @if (is_array($steps) || is_object($steps))
        @foreach($steps as $step)
            <div class="d-flex" wire:key="{{$step}}">
                <input type="text" placeholder="{{'رقم البند ' . ($step+1)}}"
                       class="form-control mb-2"
                       wire:model.defer="sub_terms.{{ $step }}">
                <span class="fa fa-times-circle text-danger p-2"
                      wire:click="remove({{$step}})"></span>
            </div>
        @endforeach
    @endif
</div>
```

**Component**

```
<?php

namespace App\Http\Livewire\Arrests;

use App\Models\Arrest;
use App\Models\ArrestedPrisoner;
use App\Models\SubTerms;
use Illuminate\Support\Facades\Validator;
use Livewire\Component;
use Livewire\WithPagination;

class ArrestedPrisoners extends Component
{
    use WithPagination;

    protected $paginationTheme = 'bootstrap';

    public $state = [];
    public $file_number;
    public $arrest_id;
    public $arrested;
    public $removeArrested = null;
    public $showEditModal = false;
    public $steps = [];
    public $sub_terms = [];


    public function increment()
    {
        $this->steps[] = count($this->steps);
    }

    public function remove($index)
    {
        unset($this->steps[$index]);
    }

    public function mount($id1, $id2)
    {
        $this->file_number = $id1;
        $this->arrest_id = $id2;

    }

    public function addNew()
    {
        $this->showEditModal = false;
        $this->state = [];
        $this->steps = [];
        $this->sub_terms = [];
        $this->dispatchBrowserEvent('show-form');
    }

    public function createArrested()
    {

        $validatedData = Validator::make($this->state, [
            'lawyer_type' => 'required',
            'current_lawyer_name' => 'required',
            'former_lawyer_name' => 'required',
            'power_attorney_date' => 'required',
            'court' => 'required',
            'current_prison' => 'required',
            'power_attorney_prison' => 'required',
            'case_size' => 'required',
            'main_concern' => 'required',
            'indictment' => 'nullable',
        ])->validate();

        $validatedData['file_number'] = $this->file_number;

        $validatedData['arrest_id'] = $this->arrest_id;

        $validatedData['terms'] = $this->sub_terms;

        $arrested_first = ArrestedPrisoner::create($validatedData);

        foreach ($validatedData['terms'] as $term) {
            if ($term != null) {
                $arrested_first->terms()->create(['terms' => $term]);
            }
        }
    }

    public function edit(ArrestedPrisoner $arrested)
    {

        foreach ($arrested['terms'] as $term) {
            $this->sub_terms = [$term->terms];
        }

        $this->steps = count($arrested['terms']);

        $this->showEditModal = true;

        $this->arrested = $arrested;

        $this->state = $arrested->toArray();

        $this->dispatchBrowserEvent('show-form');

    }

    public function updateArrested()
    {
        $validatedData = Validator::make($this->state, [
            'lawyer_type' => 'required',
            'current_lawyer_name' => 'required',
            'former_lawyer_name' => 'required',
            'power_attorney_date' => 'required',
            'court' => 'required',
            'current_prison' => 'required',
            'power_attorney_prison' => 'required',
            'case_size' => 'required',
            'main_concern' => 'required',
            'indictment' => 'nullable',
        ])->validate();
        $this->arrested->toQuery()->update($validatedData);
        $this->dispatchBrowserEvent('hide-form', ['message' => 'تم التعديل بنجاح!']);
    }

    public function confirmDeleteArrested($id)
    {
        $this->removeArrested = $id;

        $this->dispatchBrowserEvent('show_delete_modal');
    }

    public function deleteArrested()
    {
        $arrested = ArrestedPrisoner::findOrFail($this->removeArrested);

        $arrested->delete();
        $this->dispatchBrowserEvent('hide_delete_modal', ['message' => 'تم الحذف بنجاح!']);
    }

    public function render()
    {
        $this->arrested = ArrestedPrisoner::with('arrest.prisoner')
            ->where('file_number', $this->file_number)
            ->where('arrest_id', $this->arrest_id)
            ->get();
        return view('livewire.arrests.arrested-prisoners', compact($this->arrested));
    }
}


```

**Reply by ralphjsmit** [reply]
I can't test it, but one thing that is rather peculiar is that in the `edit()` function you assign an integer to the variable $steps, whereas in your Blade file you expect it to be an array or an object. Which would result in something like:

```Blade
@foreach(4 as $step)
   ///
@endforeach
```
See? This totally doesn't make sense.

Could you perhaps dump out the initial value of `$steps` or what structure it should have?

**Reply by AsemAbbas** [reply]
### I want return how many input should I have to show the value inside it but I cant make it
